### DISCUSSION TITLE
How to expose the database classes without directly importing script?

### DISCUSSION THREAD

**Discussion by mojago** [simolus3/drift#1308] [Q&A] [answered]
Hello!

First up, a massive thank you for creating and maintaining Moor - it's awesome and just what was needed for my app.

With all that said, I need a little guidance with best practices.

My database is fully working by using the get_it package to expose my database instance as a Singleton and it's working lovely.

However, I find myself importing the Moor database script (which has the MyDatabase class) into any script where I need to use the table classes or do Type checks.

Is there a better way to expose the database classes without directly importing my database script into each script where the classes are required? Or is this just an inevitability?

Thanks

**Comment by simolus3** [comment, marked as answer]
> First up, a massive thank you for creating and maintaining Moor - it's awesome and just what was needed for my app.

You're welcome, and thanks!

> However, I find myself importing the Moor database script (which has the MyDatabase class) into any script where I need to use the table classes or do Type checks.

Unfortunately, moor currently generates a part file that is inseparable from the Dart file where you define the database. I intend to change this in the future, but it will be a breaking change that requires a lot of work in the generator.

You could write another file that only exports the relevant classes from your database file:

```dart
// let's say this file is tables.dart

export 'database_file.dart' hide MyDatabaseClass;
```

Then you could import `tables.dart` elsewhere to avoid having everything in scope. Is there any reason why you're trying to avoid importing the database everywhere? If it's just for "aesthetic" reasons, e.g. because you don't want the database class to be available in a UI layer, then I think the export strategy would work.

**Reply by mojago** [reply]
> export 'database_file.dart' hide MyDatabaseClass;

Thanks for that Simon! I didn’t know you could hide classes like this, that’s super useful and will do the job in this case and others.

> Is there any reason why you're trying to avoid importing the database everywhere?

I was just checking that there wasn't some other better way of doing this to get a bit more encapsulation and potential separation.

Simplicity wins the day (yet again)!
